### PR TITLE
20: Patterns II: HOCs

### PR DESCRIPTION
**Pull request by gabrielizalo**
# Summary

* The app now includes a Loading HOC for components that uses API.

# Trello Card

https://trello.com/c/B5BC195l/20-patterns-ii-hocs

# Screenshot

loading login

loading profile

### CODE REVIEW

File: react/react-training/src/app/components/HOCLoading/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+import Loading from './components/Loading';
+
+function HOCLoading(MyComponent) {
+  return function myLoading(props) {
+    return <div>{props.isLoading ? <Loading /> : <MyComponent {...props} />}</div>;
```
Comment (henryzarza) [inline react/react-training/src/app/components/HOCLoading/index.js]: It's necessary the `propTypes` and it isn't necessary `div` use `Fragment`

File: react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js [hunk 2]
```diff
@@ -88,7 +88,11 @@ const mapDispatchToProps = dispatch => ({
   userPatch: (userId, userData) => dispatch(Actions.userPatch(userId, userData))
 });
 
-export default connect(
+const myLayoutWithLoading = HOCLoading(Layout);
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js]: Change this name for: `LayoutWithLoading`

File: react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js [hunk 3]
```diff
@@ -20,9 +21,7 @@ class Layout extends Component {
 
   render() {
     const { userData } = this.props;
-    if (!userData) {
-      return <div className={styles.loader}>Loading...</div>;
-    }
+    if (!userData) return null;
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:24]: This is necessary?

File: react/react-training/src/app/screens/Login/index.js [hunk 4]
```diff
@@ -36,7 +38,11 @@ const mapDispatchToProps = dispatch => ({
   userLoginVerify: () => dispatch(Actions.userLoginVerify())
 });
 
-export default connect(
+const myLoginWithLoading = HOCLoading(SignInForm);
```
Comment (henryzarza) [inline react/react-training/src/app/screens/Login/index.js]: Change this name for `LoginWithLoading`

**Review by henryzarza — CHANGES_REQUESTED** [review]
Excellent 👍

File: react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js (hunk 3 above)
Comment (gabrielizalo) [reply react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:24]: Yessss... Without this line I'm getting and error. 😿

**Review by henryzarza — APPROVED** [review]

**Review by wfolini — APPROVED** [review]
Congratulations!! 👏 🎉🎉

**Comment by gabrielizalo** [comment]
😄 Thankssss

**Pull request merged**